pkg/controllermanager/controller/managedseedset/managedseedset_replica.go

```diff
@@ -0,0 +1,473 @@
+	return seed.Generation == seed.Status.ObservedGeneration && seed.DeletionTimestamp == nil &&
+		conditionGardenletReady != nil && conditionGardenletReady.Status == gardencorev1beta1.ConditionTrue &&
+		conditionBootstrapped != nil && conditionBootstrapped.Status == gardencorev1beta1.ConditionTrue &&
+		(conditionBackupBucketsReady == nil || conditionBackupBucketsReady.Status == gardencorev1beta1.ConditionTrue)
+}
+
+func shootHealthStatus(shoot *gardencorev1beta1.Shoot) operationshoot.Status {
+	if value, ok := shoot.Labels[common.ShootStatus]; ok {
+		return operationshoot.Status(value)
+	}
+	return operationshoot.StatusProgressing
+}
+
+// newShoot creates a new shoot object for the given set and ordinal.
+func newShoot(set *seedmanagementv1alpha1.ManagedSeedSet, ordinal int) *gardencorev1beta1.Shoot {
+	name := getName(set, ordinal)
+
+	// Initialize shoot
+	shoot := &gardencorev1beta1.Shoot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   set.Namespace,
+			Labels:      set.Spec.ShootTemplate.Labels,
+			Annotations: set.Spec.ShootTemplate.Annotations,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(set, seedmanagementv1alpha1.SchemeGroupVersion.WithKind("ManagedSeedSet")),
+			},
+		},
+		Spec: set.Spec.ShootTemplate.Spec,
+	}
+
+	// Replace placeholders in shoot spec with the actual replica name
+	replacePlaceholdersInShootSpec(&shoot.Spec, name)
+
+	return shoot
+}
+
+// newManagedSeed creates a new managed seed object for the given set and ordinal.
+func newManagedSeed(set *seedmanagementv1alpha1.ManagedSeedSet, ordinal int) (*seedmanagementv1alpha1.ManagedSeed, error) {
+	name := getName(set, ordinal)
+
+	// Initialize managed seed
+	managedSeed := &seedmanagementv1alpha1.ManagedSeed{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   set.Namespace,
+			Labels:      set.Spec.Template.Labels,
+			Annotations: set.Spec.Template.Annotations,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(set, seedmanagementv1alpha1.SchemeGroupVersion.WithKind("ManagedSeedSet")),
+			},
+		},
+		Spec: seedmanagementv1alpha1.ManagedSeedSpec{
+			Shoot: &seedmanagementv1alpha1.Shoot{
+				Name: name,
+			},
+			SeedTemplate: set.Spec.Template.Spec.SeedTemplate,
+			Gardenlet:    set.Spec.Template.Spec.Gardenlet,
+		},
+	}
+
+	// Replace placeholders in seed spec with the actual replica name
+	switch {
+	case managedSeed.Spec.SeedTemplate != nil:
+		replacePlaceholdersInSeedSpec(&managedSeed.Spec.SeedTemplate.Spec, name)
+	case managedSeed.Spec.Gardenlet != nil:
+		// Decode gardenlet configuration
+		gardenletConfig, err := seedmanagementhelper.DecodeGardenletConfiguration(&managedSeed.Spec.Gardenlet.Config, false)
+		if err != nil {
+			return nil, err
+		}
+		replacePlaceholdersInSeedSpec(&gardenletConfig.SeedConfig.Spec, name)
+	}
+
+	return managedSeed, nil
+}
+
+const placeholder = "replica-name"
```

Comment:
Honestly, I don't see why we should introduce such an artificial limitation when we could just use a placeholder for the replica name as already implemented. Not to mention that this solution is also easier to implement and maintain. What's actually wrong with it so you insist on changing it?

Reply:
Replacing some magical string in a resource template just feels wrong to me and I can't think of an example where kubernetes does this.
I wanted to raise this here to get some more opinions (maybe I'm alone with my opinion, I don't know...)
@gardener/gardener-maintainers, any preferences?

Reply:
Perhaps kubernetes doesn't do it because they simply don't have resources similar to this one, where the generated replica name could be used in some non-trivial way inside the template. We however do have this case here. A "magical string" would be less magical if properly documented.

Reply:
I understand both arguments, though, I would also rather tend to simplify the UX here. I see that using the placeholder has the benefit that the domain can be chosen arbitrarily, but I'm not sure if this is so essential that it justifies having such "magic strings". OTOH, @stoyanr is right that we would impose artificial limitations otherwise.

Generally, I don't like that this is not "optional", i.e., if I want to specify the domain then I have to know about this "magic string" and what happens when I use it. It's not very complex, but still, if I would have implemented it then I would probably forbid specifying both `.spec.dns.domain` in the `Shoot` and the `.spec.{dns.ingressDomain,ingress.domain}` fields in the `Seed` and add a separate field `.spec.baseDomain` in the `ManagedSeedSet`. Then the `Shoot` domain can be prepended with the `<replica-name>` and the `.spec.ingress.domain` can be prepended with `ingress.<replica-name>` as suggested by @timebertt.
This has the benefit that the operator is not bothered with these placeholder details, but obviously, it limits him using domains with other prefixes. For me personally, the improved UX prevails here.

Btw, I would always use the managed ingress for `ManagedSeedSet`s and forbid specifying `.spec.dns.ingressDomain` in general (independent of above question).

Btw, can you rather use `example.com` instead of `ondemand.com`?

Reply:
@rfranzke We took another look at the implications together with @timebertt and jointly decided in favor of the placeholder approach. The main reason for this is that if we try to allow empty DNS domain in either seed or shoot templates, in some cases (e.g. custom providers), this would also fail the validation, which would complicate things even further - we would either need to introduce further limitations, or to find some way around these validation failures. 

By comparison, the placeholder approach requires no limitations and can be implemented with the least amount of code / changes to existing code. If we have a strong reason we could still change it in the future I believe.